mlrun/common/model_monitoring/helpers.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import mlrun.api
+import mlrun.common
+from mlrun.common.schemas.model_monitoring import (
+    EndpointUID,
+    FunctionURI,
+    VersionedModel,
+)
+from mlrun.config import is_running_as_api
+
+
+def create_model_endpoint_uid(function_uri: str, versioned_model: str):
+    function_uri = FunctionURI.from_string(function_uri)
+    versioned_model = VersionedModel.from_string(versioned_model)
+
+    if (
+        not function_uri.project
+        or not function_uri.function
+        or not versioned_model.model
+    ):
+        raise ValueError("Both function_uri and versioned_model have to be initialized")
+
+    uid = EndpointUID(
+        function_uri.project,
+        function_uri.function,
+        function_uri.tag,
+        function_uri.hash_key,
+        versioned_model.model,
+        versioned_model.version,
+    )
+
+    return uid
+
+
+def parse_model_endpoint_project_prefix(path: str, project_name: str):
+    return path.split(project_name, 1)[0] + project_name
+
+
+def parse_model_endpoint_store_prefix(store_prefix: str):
+    endpoint, parsed_url = mlrun.platforms.iguazio.parse_path(store_prefix)
+    container, path = parsed_url.split("/", 1)
+    return endpoint, container, path
+
+
+def get_stream_path(project: str = None):
+    # TODO: This function (as well as other methods in this file) includes both client and server side code. We will
```

Comment:
why can't we two seperate functions for client and server?

Reply:
We wanted to keep consistency with `get_connection_string()` but actually in the `get_stream_path()` case we can separate it for client and server. Therefore, we have now two new `get_stream_path()` functions instead of the old one:
- `mlrun/model_monitoring/helpers.py` (Client)
- `mlrun/api/crud/model_monitoring/helpers.py` (API)